onmetal-controller-manager/controllers/networking/loadbalancer_controller.go

```diff
@@ -0,0 +1,188 @@
+	destinations, err := r.findDestinations(ctx, log, loadBalancer)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("error finding destinations: %w", err)
+	}
+	log.V(1).Info("Successfully found destinations", "Destinations", destinations)
+
+	log.V(1).Info("Applying routing")
+	if err := r.applyRouting(ctx, loadBalancer, destinations); err != nil {
+		return ctrl.Result{}, fmt.Errorf("error applying routing: %w", err)
+	}
+	log.V(1).Info("Successfully applied routing")
+	return ctrl.Result{}, nil
+}
+
+func (r *LoadBalancerReconciler) findDestinations(ctx context.Context, log logr.Logger, loadBalancer *networkingv1alpha1.LoadBalancer) ([]commonv1alpha1.LocalUIDReference, error) {
+	sel, err := metav1.LabelSelectorAsSelector(loadBalancer.Spec.NetworkInterfaceSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	nicList := &networkingv1alpha1.NetworkInterfaceList{}
+	if err := r.List(ctx, nicList,
+		client.InNamespace(loadBalancer.Namespace),
+		client.MatchingLabelsSelector{Selector: sel},
+		client.MatchingFields{onmetalapiclient.NetworkInterfaceNetworkName: loadBalancer.Spec.NetworkRef.Name},
+	); err != nil {
+		return nil, fmt.Errorf("error listing network interfaces: %w", err)
+	}
+
+	destinations := make([]commonv1alpha1.LocalUIDReference, 0, len(nicList.Items))
+	for _, nic := range nicList.Items {
+		destinations = append(destinations, commonv1alpha1.LocalUIDReference{Name: nic.Name, UID: nic.UID})
+	}
+	return destinations, nil
+}
+
+func (r *LoadBalancerReconciler) applyRouting(ctx context.Context, loadBalancer *networkingv1alpha1.LoadBalancer, destinations []commonv1alpha1.LocalUIDReference) error {
+	loadBalancerRouting := &networkingv1alpha1.LoadBalancerRouting{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "LoadBalancerRouting",
+			APIVersion: networkingv1alpha1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: loadBalancer.Namespace,
+			Name:      loadBalancer.Name,
+		},
+		Destinations: destinations,
+	}
+	if err := ctrl.SetControllerReference(loadBalancer, loadBalancerRouting, r.Scheme); err != nil {
+		return fmt.Errorf("error setting controller reference: %w", err)
+	}
+	if err := r.Patch(ctx, loadBalancerRouting, client.Apply, loadBalancerFieldOwner, client.ForceOwnership); err != nil {
+		return fmt.Errorf("error applying loadbalancer routing: %w", err)
+	}
+	return nil
+}
+
+func (r *LoadBalancerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ctx := context.Background()
+	log := ctrl.Log.WithName("loadbalancer").WithName("setup")
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&networkingv1alpha1.LoadBalancer{}).
+		Owns(&networkingv1alpha1.LoadBalancerRouting{}).
+		Watches(
+			&source.Kind{Type: &networkingv1alpha1.NetworkInterface{}},
+			r.enqueueByLoadBalancerMatchingNetworkInterface(log, ctx),
+		).
+		Complete(r)
+}
+
+func (r *LoadBalancerReconciler) enqueueByLoadBalancerMatchingNetworkInterface(log logr.Logger, ctx context.Context) handler.EventHandler {
```

Comment:
`ctx` should go first in a method (yes, might have happened, that it's in some of our packages, but that's wrong as well 🙈 )